ed_sensor_integration/tools/fitter_viz.cpp

```diff
@@ -307,197 +354,31 @@ int main(int argc, char **argv)
         cv::Scalar sensorcolor(0,255,0); // green
         cv::circle(canvas, sensorlocation, 3, sensorcolor, cv::FILLED);
 
-    // paint entity (from worldmodel)
-        EntityRepresentation2D entity_2d = fitter.GetOrCreateEntity2D(e);
-
-        geo::Pose3D pose = e->pose();
-
-        // worldmodel entity yaw (unfitted)
-        float yaw_ent = getYaw(pose.R);
-
-        // robot yaw
-        float yaw_robot = getYaw(snapshot.sensor_pose.R);
-
-        for (int i=0; i < entity_2d.shape_2d.size(); i++){
-
-            for (int j=0; j < entity_2d.shape_2d[i].size()-1; j++){
-                // computing outer corner positions (in worldmodel coordinate system)
-                float x_wm1 = e->pose().t.x + ( cos(yaw_ent) * entity_2d.shape_2d[i][j].x - sin(yaw_ent) * entity_2d.shape_2d[i][j].y);
-                float y_wm1 = e->pose().t.y + ( sin(yaw_ent) * entity_2d.shape_2d[i][j].x + cos(yaw_ent) * entity_2d.shape_2d[i][j].y);
-                float x_wm2 = e->pose().t.x + ( cos(yaw_ent) * entity_2d.shape_2d[i][j+1].x - sin(yaw_ent) * entity_2d.shape_2d[i][j+1].y);
-                float y_wm2 = e->pose().t.y + ( sin(yaw_ent) * entity_2d.shape_2d[i][j+1].x + cos(yaw_ent) * entity_2d.shape_2d[i][j+1].y);
+        geo::Transform2 sensor_pose2d(fitterdata.sensor_pose_xya.t.x, fitterdata.sensor_pose_xya.t.y, getYaw(fitterdata.sensor_pose_xya.R));
+        geo::Transform2 entity_pose2d(e->pose().t.x, e->pose().t.y, getYaw(e->pose().R));
+        geo::Transform2 fitted_pose2d(fitted_pose.t.x, fitted_pose.t.y, getYaw(fitted_pose.R));
```

Comment:
I think we can use the [`projectTo2d`](https://tue-robotics.github.io/docs/geolib2/html/classgeo_1_1Transform3T.html#a57186a12e9cc570e7cdfbf03c5b1fab2) function here.